src/Controllers/controller.py

```diff
@@ -2,18 +2,59 @@
 from Controllers.cli_controller import CLI_Controller
 from Controllers.gui_controller import GUI_Controller
 from Views.cli_view import CLI_View
+from prompt_toolkit import prompt
+from prompt_toolkit.completion import NestedCompleter
 class UML_Controller:
 
     def __init__(self):
+        """
+        Initializes the instance with either a CLI_Controller or GUI_Controller based on user choice, 
+        and sets up autocomplete functionality.
+        """
         self._controller:CLI_Controller | GUI_Controller = self.__pick_controller()
+        self.setup_autocomplete()  
         
     
     def run(self):
+        """Executes the main loop of the program, utilizing the prompt_toolkit prompt for autocomplete."""
         while not self._controller._should_quit:
-            self._controller.update(input("Command: ").strip().lower())
-    
+            self._controller.update(prompt("Command: ", completer=self._completer).strip().lower())
+
     def __pick_controller(self, args:str = sys.argv) -> CLI_Controller | GUI_Controller: 
         if len(args) > 1 and str(args[1]).strip().lower() == 'cli':
             return CLI_Controller()
         return GUI_Controller()
+    
+    def setup_autocomplete(self):
```

Comment:
sorry, I didn't mean to imply that it needed to be moved. It was fine where it was I just didn't know how to actually invoke it vs just prompting like regular.